Installation (from scratch, no migration) gives "no user found"

Trying to perform an installation from scratch (thus no migration) gives "no user found", see also screenshot and extract from log file below.

Information about the deployment:

- Ubuntu v20.04LTS
- Java 1.8.0_292-8u292-b10-0ubuntu1~20.04-b10
- Apache Tomcat 8.5.70
- Apache Solr 8.9.0
- Apache Server installed, but the application on the Tomcat is accessed directly
- Apache Shiro is used
- the code has been modified at some points, see https://github.com/heidivanparys/re3gistry/commits/additional_changes_2, mainly to 
  - use Oracle instead of PostgreSQL
  - add trace and debug logging
  - fix some issues encountered earlier, see also previously created pull requests
  - to make more properties configurable via Maven profiles

Investigation so far:

The code sets an attribute with the newly created administrator user on the session:

https://github.com/ec-jrc/re3gistry/blob/79f806053066cfa1fb1636ce34b64912004ccb32/sources/Re3gistry2JavaAPI/src/main/java/eu/europa/ec/re3gistry2/javaapi/handler/RegInstallationStep3Handler.java#L62

but in the next step, no user is found on the session (because it is in fact another session?), resulting in an error:

https://github.com/ec-jrc/re3gistry/blob/79f806053066cfa1fb1636ce34b64912004ccb32/sources/Re3gistry2JavaAPI/src/main/java/eu/europa/ec/re3gistry2/javaapi/handler/RegInstallationStepCleanInstallationProcessHandler.java#L75-L79


![image](https://user-images.githubusercontent.com/11427611/141021776-cfca2680-76a2-44a7-8184-7d5d33323ac9.png)

Any help on how to solve this issue would be greatly appreciated.

```
[TRACE] 2021-11-10 00:18:40.379 [http-nio-8080-exec-173] [controller.LoadSalt.doFilter:52] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[TRACE] 2021-11-10 00:18:40.392 [http-nio-8080-exec-173] [controller.LoadSalt.doFilter:60] - Creating CSRF prevention salt cache
[TRACE] 2021-11-10 00:18:40.415 [http-nio-8080-exec-173] [controller.LoadSalt.doFilter:69] - Generate the salt, store it in the users cache, and add it to the current request
[TRACE] 2021-11-10 00:18:40.419 [http-nio-8080-exec-173] [controller.ValidateSalt.doFilter:50] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[DEBUG] 2021-11-10 00:18:40.419 [http-nio-8080-exec-173] [controller.ValidateSalt.doFilter:58] - Validating that the salt is in the cache
[DEBUG] 2021-11-10 00:18:40.419 [http-nio-8080-exec-173] [controller.ValidateSalt.doFilter:66] - Salt is in cache, thus moving on
[TRACE] 2021-11-10 00:18:40.432 [http-nio-8080-exec-173] [controller.Install.processRequest:54] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-11-10 00:18:40.433 [http-nio-8080-exec-173] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-11-10 00:18:40.433 [http-nio-8080-exec-173] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-11-10 00:18:40.433 [http-nio-8080-exec-173] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-11-10 00:18:40.433 [http-nio-8080-exec-173] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-11-10 00:18:40.433 [http-nio-8080-exec-173] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-11-10 00:18:40.434 [http-nio-8080-exec-173] [utility.Configuration.initServlet:205] - Current page name=/install
[TRACE] 2021-11-10 00:18:40.434 [http-nio-8080-exec-173] [utility.Configuration.initServlet:213] - No user found in the session, checking user
[TRACE] 2021-11-10 00:18:42.327 [http-nio-8080-exec-173] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-11-10 00:18:42.328 [http-nio-8080-exec-173] [utility.Configuration.initServlet:221] - No user found
[TRACE] 2021-11-10 00:18:42.341 [http-nio-8080-exec-173] [controller.Install.processRequest:64] - Session id: 88884FC5F0ECD8961C899BB7B3C47123
[DEBUG] 2021-11-10 00:18:42.342 [http-nio-8080-exec-173] [controller.Install.processRequest:74] - install step = 1
[TRACE] 2021-11-10 00:18:48.101 [http-nio-8080-exec-174] [controller.LoadSalt.doFilter:52] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[TRACE] 2021-11-10 00:18:48.102 [http-nio-8080-exec-174] [controller.LoadSalt.doFilter:60] - Creating CSRF prevention salt cache
[TRACE] 2021-11-10 00:18:48.102 [http-nio-8080-exec-174] [controller.LoadSalt.doFilter:69] - Generate the salt, store it in the users cache, and add it to the current request
[TRACE] 2021-11-10 00:18:48.103 [http-nio-8080-exec-174] [controller.ValidateSalt.doFilter:50] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[DEBUG] 2021-11-10 00:18:48.103 [http-nio-8080-exec-174] [controller.ValidateSalt.doFilter:58] - Validating that the salt is in the cache
[DEBUG] 2021-11-10 00:18:48.103 [http-nio-8080-exec-174] [controller.ValidateSalt.doFilter:66] - Salt is in cache, thus moving on
[TRACE] 2021-11-10 00:18:48.104 [http-nio-8080-exec-174] [controller.Install.processRequest:54] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-11-10 00:18:48.104 [http-nio-8080-exec-174] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-11-10 00:18:48.104 [http-nio-8080-exec-174] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-11-10 00:18:48.105 [http-nio-8080-exec-174] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-11-10 00:18:48.105 [http-nio-8080-exec-174] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-11-10 00:18:48.105 [http-nio-8080-exec-174] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-11-10 00:18:48.105 [http-nio-8080-exec-174] [utility.Configuration.initServlet:205] - Current page name=/install
[TRACE] 2021-11-10 00:18:48.106 [http-nio-8080-exec-174] [utility.Configuration.initServlet:213] - No user found in the session, checking user
[TRACE] 2021-11-10 00:18:48.132 [http-nio-8080-exec-174] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-11-10 00:18:48.132 [http-nio-8080-exec-174] [utility.Configuration.initServlet:221] - No user found
[TRACE] 2021-11-10 00:18:48.146 [http-nio-8080-exec-174] [controller.Install.processRequest:64] - Session id: 4C933354ED0B8B9D1022ECFA9360356C
[DEBUG] 2021-11-10 00:18:48.222 [http-nio-8080-exec-174] [controller.Install.processRequest:74] - install step = 2
[TRACE] 2021-11-10 00:19:01.423 [http-nio-8080-exec-170] [controller.LoadSalt.doFilter:52] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[TRACE] 2021-11-10 00:19:01.423 [http-nio-8080-exec-170] [controller.LoadSalt.doFilter:60] - Creating CSRF prevention salt cache
[TRACE] 2021-11-10 00:19:01.423 [http-nio-8080-exec-170] [controller.LoadSalt.doFilter:69] - Generate the salt, store it in the users cache, and add it to the current request
[TRACE] 2021-11-10 00:19:01.424 [http-nio-8080-exec-170] [controller.ValidateSalt.doFilter:50] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[DEBUG] 2021-11-10 00:19:01.424 [http-nio-8080-exec-170] [controller.ValidateSalt.doFilter:58] - Validating that the salt is in the cache
[DEBUG] 2021-11-10 00:19:01.424 [http-nio-8080-exec-170] [controller.ValidateSalt.doFilter:66] - Salt is in cache, thus moving on
[TRACE] 2021-11-10 00:19:01.425 [http-nio-8080-exec-170] [controller.Install.processRequest:54] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-11-10 00:19:01.425 [http-nio-8080-exec-170] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-11-10 00:19:01.425 [http-nio-8080-exec-170] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-11-10 00:19:01.426 [http-nio-8080-exec-170] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-11-10 00:19:01.426 [http-nio-8080-exec-170] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-11-10 00:19:01.426 [http-nio-8080-exec-170] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-11-10 00:19:01.426 [http-nio-8080-exec-170] [utility.Configuration.initServlet:205] - Current page name=/install
[TRACE] 2021-11-10 00:19:01.427 [http-nio-8080-exec-170] [utility.Configuration.initServlet:213] - No user found in the session, checking user
[TRACE] 2021-11-10 00:19:01.455 [http-nio-8080-exec-170] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-11-10 00:19:01.455 [http-nio-8080-exec-170] [utility.Configuration.initServlet:221] - No user found
[TRACE] 2021-11-10 00:19:01.467 [http-nio-8080-exec-170] [controller.Install.processRequest:64] - Session id: A2A11E01D46DBD7F3450BD9599F1AC39
[DEBUG] 2021-11-10 00:19:01.467 [http-nio-8080-exec-170] [controller.Install.processRequest:74] - install step = 3
[DEBUG] 2021-11-10 00:19:01.482 [http-nio-8080-exec-170] [handler.RegInstallationHandler.createRegUser:602] - Creating new administrator user
[DEBUG] 2021-11-10 00:19:01.487 [http-nio-8080-exec-170] [handler.RegInstallationHandler.createRegUser:622] - Administrator user eu.europa.ec.re3gistry2.model.RegUser[ uuid=4ea865adface3494971c6deb5ab1deac ] created
[DEBUG] 2021-11-10 00:19:01.608 [http-nio-8080-exec-170] [handler.RegInstallationHandler.createRegUser:631] - Administrator user eu.europa.ec.re3gistry2.model.RegUser[ uuid=4ea865adface3494971c6deb5ab1deac ] persisted and committed
[DEBUG] 2021-11-10 00:19:01.721 [http-nio-8080-exec-170] [handler.RegInstallationStep3Handler.addUser:64] - Set user eu.europa.ec.re3gistry2.model.RegUser[ uuid=4ea865adface3494971c6deb5ab1deac ] as an attribute on the session
[TRACE] 2021-11-10 00:19:04.363 [http-nio-8080-exec-164] [controller.LoadSalt.doFilter:52] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[TRACE] 2021-11-10 00:19:04.363 [http-nio-8080-exec-164] [controller.LoadSalt.doFilter:60] - Creating CSRF prevention salt cache
[TRACE] 2021-11-10 00:19:04.363 [http-nio-8080-exec-164] [controller.LoadSalt.doFilter:69] - Generate the salt, store it in the users cache, and add it to the current request
[TRACE] 2021-11-10 00:19:04.363 [http-nio-8080-exec-164] [controller.ValidateSalt.doFilter:50] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[DEBUG] 2021-11-10 00:19:04.364 [http-nio-8080-exec-164] [controller.ValidateSalt.doFilter:58] - Validating that the salt is in the cache
[DEBUG] 2021-11-10 00:19:04.364 [http-nio-8080-exec-164] [controller.ValidateSalt.doFilter:66] - Salt is in cache, thus moving on
[TRACE] 2021-11-10 00:19:04.365 [http-nio-8080-exec-164] [controller.Install.processRequest:54] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-11-10 00:19:04.365 [http-nio-8080-exec-164] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-11-10 00:19:04.365 [http-nio-8080-exec-164] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-11-10 00:19:04.365 [http-nio-8080-exec-164] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-11-10 00:19:04.366 [http-nio-8080-exec-164] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-11-10 00:19:04.366 [http-nio-8080-exec-164] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-11-10 00:19:04.366 [http-nio-8080-exec-164] [utility.Configuration.initServlet:205] - Current page name=/install
[TRACE] 2021-11-10 00:19:04.366 [http-nio-8080-exec-164] [utility.Configuration.initServlet:213] - No user found in the session, checking user
[TRACE] 2021-11-10 00:19:04.382 [http-nio-8080-exec-164] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-11-10 00:19:04.383 [http-nio-8080-exec-164] [utility.Configuration.initServlet:221] - No user found
[TRACE] 2021-11-10 00:19:04.394 [http-nio-8080-exec-164] [controller.Install.processRequest:64] - Session id: 37FC2D1C38C865EC42B307E8FF4FE1C3
[DEBUG] 2021-11-10 00:19:04.395 [http-nio-8080-exec-164] [controller.Install.processRequest:74] - install step = clean-installation
[TRACE] 2021-11-10 00:19:47.279 [http-nio-8080-exec-176] [controller.LoadSalt.doFilter:52] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[TRACE] 2021-11-10 00:19:47.279 [http-nio-8080-exec-176] [controller.LoadSalt.doFilter:60] - Creating CSRF prevention salt cache
[TRACE] 2021-11-10 00:19:47.280 [http-nio-8080-exec-176] [controller.LoadSalt.doFilter:69] - Generate the salt, store it in the users cache, and add it to the current request
[TRACE] 2021-11-10 00:19:47.280 [http-nio-8080-exec-176] [controller.ValidateSalt.doFilter:50] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[DEBUG] 2021-11-10 00:19:47.280 [http-nio-8080-exec-176] [controller.ValidateSalt.doFilter:58] - Validating that the salt is in the cache
[DEBUG] 2021-11-10 00:19:47.280 [http-nio-8080-exec-176] [controller.ValidateSalt.doFilter:66] - Salt is in cache, thus moving on
[TRACE] 2021-11-10 00:19:47.281 [http-nio-8080-exec-176] [controller.Install.processRequest:54] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-11-10 00:19:47.281 [http-nio-8080-exec-176] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-11-10 00:19:47.281 [http-nio-8080-exec-176] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-11-10 00:19:47.282 [http-nio-8080-exec-176] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-11-10 00:19:47.282 [http-nio-8080-exec-176] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-11-10 00:19:47.282 [http-nio-8080-exec-176] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-11-10 00:19:47.283 [http-nio-8080-exec-176] [utility.Configuration.initServlet:205] - Current page name=/install
[TRACE] 2021-11-10 00:19:47.283 [http-nio-8080-exec-176] [utility.Configuration.initServlet:213] - No user found in the session, checking user
[TRACE] 2021-11-10 00:19:47.308 [http-nio-8080-exec-176] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-11-10 00:19:47.308 [http-nio-8080-exec-176] [utility.Configuration.initServlet:221] - No user found
[TRACE] 2021-11-10 00:19:47.320 [http-nio-8080-exec-176] [controller.Install.processRequest:64] - Session id: A6D887CB56C1EA19534555744763D60B
[DEBUG] 2021-11-10 00:19:47.320 [http-nio-8080-exec-176] [controller.Install.processRequest:74] - install step = clean-installation-summary
[TRACE] 2021-11-10 00:19:58.745 [http-nio-8080-exec-173] [controller.LoadSalt.doFilter:52] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[TRACE] 2021-11-10 00:19:58.746 [http-nio-8080-exec-173] [controller.LoadSalt.doFilter:60] - Creating CSRF prevention salt cache
[TRACE] 2021-11-10 00:19:58.746 [http-nio-8080-exec-173] [controller.LoadSalt.doFilter:69] - Generate the salt, store it in the users cache, and add it to the current request
[TRACE] 2021-11-10 00:19:58.747 [http-nio-8080-exec-173] [controller.ValidateSalt.doFilter:50] - Start filter eu.europa.ec.re3gistry2.web.controller.ValidateSalt
[DEBUG] 2021-11-10 00:19:58.747 [http-nio-8080-exec-173] [controller.ValidateSalt.doFilter:58] - Validating that the salt is in the cache
[DEBUG] 2021-11-10 00:19:58.747 [http-nio-8080-exec-173] [controller.ValidateSalt.doFilter:66] - Salt is in cache, thus moving on
[TRACE] 2021-11-10 00:19:58.748 [http-nio-8080-exec-173] [controller.Install.processRequest:54] - Start processing request in eu.europa.ec.re3gistry2.web.controller.Install
[DEBUG] 2021-11-10 00:19:58.748 [http-nio-8080-exec-173] [utility.Configuration.initServlet:162] - Start initialising servlet
[DEBUG] 2021-11-10 00:19:58.748 [http-nio-8080-exec-173] [utility.Configuration.initServlet:180] - Proceed initialising servlet
[TRACE] 2021-11-10 00:19:58.748 [http-nio-8080-exec-173] [utility.Configuration.initServlet:183] - Initialise properties
[TRACE] 2021-11-10 00:19:58.749 [http-nio-8080-exec-173] [utility.Configuration.initServlet:188] - Initialise localisation
[TRACE] 2021-11-10 00:19:58.749 [http-nio-8080-exec-173] [utility.Configuration.initServlet:193] - Initialise available languages in session if needed
[DEBUG] 2021-11-10 00:19:58.749 [http-nio-8080-exec-173] [utility.Configuration.initServlet:205] - Current page name=/install
[TRACE] 2021-11-10 00:19:58.749 [http-nio-8080-exec-173] [utility.Configuration.initServlet:213] - No user found in the session, checking user
[TRACE] 2021-11-10 00:19:58.770 [http-nio-8080-exec-173] [utility.UserHelper.checkUser:81] - loginType=SHIRO
[DEBUG] 2021-11-10 00:19:58.770 [http-nio-8080-exec-173] [utility.Configuration.initServlet:221] - No user found
[TRACE] 2021-11-10 00:19:58.782 [http-nio-8080-exec-173] [controller.Install.processRequest:64] - Session id: 0A70FB5F1D490AE8CCE36E1BB4A60FA6
[DEBUG] 2021-11-10 00:19:58.782 [http-nio-8080-exec-173] [controller.Install.processRequest:74] - install step = clean-installation-process
[INFO ] 2021-11-10 00:19:58.782 [http-nio-8080-exec-173] [controller.Install.processRequest:110] - Start clean installation
[ERROR] 2021-11-10 00:19:58.783 [http-nio-8080-exec-173] [controller.Install.processRequest:123] - Exception occurred in step clean-installation-process
java.lang.Exception: There is no user
	at eu.europa.ec.re3gistry2.javaapi.handler.RegInstallationStepCleanInstallationProcessHandler.createRegistryFromSession(RegInstallationStepCleanInstallationProcessHandler.java:78) ~[Re3gistry2JavaAPI-1.0.jar:?]
	at eu.europa.ec.re3gistry2.javaapi.handler.RegInstallationStepCleanInstallationProcessHandler.<init>(RegInstallationStepCleanInstallationProcessHandler.java:65) ~[Re3gistry2JavaAPI-1.0.jar:?]
	at eu.europa.ec.re3gistry2.web.controller.Install.processRequest(Install.java:113) [classes/:?]
	at eu.europa.ec.re3gistry2.web.controller.Install.doPost(Install.java:180) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681) [servlet-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:47) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:61) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AdviceFilter.executeChain(AdviceFilter.java:108) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AdviceFilter.doFilterInternal(AdviceFilter.java:137) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.ProxiedFilterChain.doFilter(ProxiedFilterChain.java:66) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:387) [shiro-core-1.8.0.jar:1.8.0]
	at org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125) [shiro-web-1.2.4.jar:1.2.4]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.ValidateSalt.doFilter(ValidateSalt.java:67) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at eu.europa.ec.re3gistry2.web.controller.LoadSalt.doFilter(LoadSalt.java:78) [classes/:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71) [log4j-web-2.14.1.jar:2.14.1]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.70]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:196) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) [catalina.jar:8.5.70]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:544) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.70]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:698) [catalina.jar:8.5.70]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) [catalina.jar:8.5.70]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:364) [catalina.jar:8.5.70]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:624) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.70]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:831) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1650) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:8.5.70]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.70]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]

```

--- a/sources/Re3gistry2-build-helper/pom.xml
+++ b/sources/Re3gistry2-build-helper/pom.xml
@@ -55,6 +55,7 @@
                 <persistence.jdbc.password>DB_PASSWORD</persistence.jdbc.password>
                 <application.rooturl>https://localhost/r3egistry2</application.rooturl> 
 				<application.releasenote.rss.path>/path/to/rss</application.releasenote.rss.path>
+				<web.session.cookie.secure>true</web.session.cookie.secure>
             </properties>
         </profile>        
         <!-- Server profile -->
@@ -69,6 +70,7 @@
                 <persistence.jdbc.password>DB_PASSWORD</persistence.jdbc.password>            
                 <application.rooturl>https://registry-test.eu/re3gistry2</application.rooturl>
                 <application.solr.isactive>false</application.solr.isactive>        
+                <web.session.cookie.secure>true</web.session.cookie.secure>
             </properties>
         </profile>     
     </profiles>

--- a/sources/Re3gistry2/pom.xml
+++ b/sources/Re3gistry2/pom.xml
@@ -75,6 +75,7 @@
                 <version>3.2.2</version>
                 <configuration>
                     <failOnMissingWebXml>false</failOnMissingWebXml>
+                    <filteringDeploymentDescriptors>true</filteringDeploymentDescriptors>
                 </configuration>
             </plugin>
             <plugin>

--- a/sources/Re3gistry2/src/main/java/eu/europa/ec/re3gistry2/web/controller/Install.java
+++ b/sources/Re3gistry2/src/main/java/eu/europa/ec/re3gistry2/web/controller/Install.java
@@ -50,6 +50,7 @@ import org.apache.logging.log4j.Logger;
 public class Install extends HttpServlet {
 
     private void processRequest(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        Configuration.getInstance().getLogger().trace("Start processing request in " + Install.class.getName());
 
         // Init frontend servlet
         Configuration.getInstance().initServlet(request, response, true, false);
@@ -67,6 +68,8 @@ public class Install extends HttpServlet {
         if (step == null || step.length() < 1) {
             step = "1";
         }
+
+        Configuration.getInstance().getLogger().debug("install step = " + step);
 
         if (step.equals("2")) {
             request.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_ADMIN_USER_ERROR, BaseConstants.KEY_REQUEST_USER_CREATION_STARTED);
@@ -95,11 +98,13 @@ public class Install extends HttpServlet {
             try {
                 new RegInstallationStepCleanInstallationSummaryHandler(request);
             } catch (Exception ex) {
+                Configuration.getInstance().getLogger().error(ex.getMessage(), ex);
                 step = BaseConstants.KEY_REQUEST_CLEAN_INSTALLATION_SUMMARY;
                 request.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_CLEAN_DB_ERROR, ex.getMessage());
             }
         }
         if (step.equals(BaseConstants.KEY_REQUEST_CLEAN_INSTALLATION_PROCESS)) {
+            Configuration.getInstance().getLogger().info("Start clean installation");
             try {
                 createSystemInstallingFile();
                 new RegInstallationStepCleanInstallationProcessHandler(request, entityManagerRe3gistry2);
@@ -110,7 +115,9 @@ public class Install extends HttpServlet {
                     session.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_SUCCESS, BaseConstants.KEY_REQUEST_INSTALLATION_SUCCESS);
                 }
                 request.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_SUCCESS, BaseConstants.KEY_REQUEST_INSTALLATION_SUCCESS);
+                Configuration.getInstance().getLogger().info("Installation succeeded");
             } catch (Exception ex) {
+                Configuration.getInstance().getLogger().error("Exception occurred in step " + BaseConstants.KEY_REQUEST_CLEAN_INSTALLATION_PROCESS, ex);
                 step = "3";
                 deleteSystemInstallingFile();
                 request.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_CLEAN_DB_ERROR, ex.getMessage());
@@ -121,6 +128,7 @@ public class Install extends HttpServlet {
             try {
                 new RegInstallationStepMigrationSummaryHandler(request, entityManagerRe3gistry2);
             } catch (Exception ex) {
+                Configuration.getInstance().getLogger().error("Exception occurred in step " + BaseConstants.KEY_REQUEST_MIGRATION_SUMMARY, ex);
                 step = "3";
                 request.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_MIGRATION_ERROR, ex.getMessage());
             }
@@ -137,6 +145,7 @@ public class Install extends HttpServlet {
                     session.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_SUCCESS, BaseConstants.KEY_REQUEST_INSTALLATION_SUCCESS);
                 }
             } catch (Exception ex) {
+                Configuration.getInstance().getLogger().error("Exception occurred in step " + BaseConstants.KEY_REQUEST_MIGRATION_PROCESS, ex);
                 step = "3";
                 deleteSystemInstallingFile();
                 request.setAttribute(BaseConstants.KEY_REQUEST_INSTALLATION_MIGRATION_ERROR, ex.getMessage());
@@ -174,19 +183,20 @@ public class Install extends HttpServlet {
 
     private void createSystemInstallingFile() throws Exception {
         String propertiesPath = System.getProperty(BaseConstants.KEY_FOLDER_NAME_CONFIGURATIONS);
-        String systemInstalledPath = propertiesPath + File.separator + BaseConstants.KEY_FILE_NAME_SYSTEMINSTALLING;
-        File systemInstalledFile = new File(systemInstalledPath);
+        String systemInstallingPath = propertiesPath + File.separator + BaseConstants.KEY_FILE_NAME_SYSTEMINSTALLING;
+        File systemInstallingFile = new File(systemInstallingPath);
 
-        systemInstalledFile.getParentFile().mkdirs();
+        systemInstallingFile.getParentFile().mkdirs();
         try {
 
-            boolean success = systemInstalledFile.createNewFile();
+            boolean success = systemInstallingFile.createNewFile();
 
             if (!success) {
                 throw new IOException();
             }
 
         } catch (IOException ex) {
+            Configuration.getInstance().getLogger().error("Error while trying to create the system installing file", ex);
             throw new Exception(ex.getMessage());
         }
     }
@@ -200,6 +210,7 @@ public class Install extends HttpServlet {
         try {
             success = file.delete();
         } catch (Exception e) {
+          Configuration.getInstance().getLogger().error("Error while trying to delete the system installing file", e);
         }
 
 //        if (!success) {
@@ -221,6 +232,7 @@ public class Install extends HttpServlet {
             }
 
         } catch (IOException ex) {
+            Configuration.getInstance().getLogger().error("Error while trying to create the system installed file", ex);
             throw new Exception(ex.getMessage());
         }
     }

--- a/sources/Re3gistry2/src/main/webapp/WEB-INF/web.xml
+++ b/sources/Re3gistry2/src/main/webapp/WEB-INF/web.xml
@@ -3,8 +3,14 @@
     <display-name>Re3gistry2</display-name>
     <session-config>
         <cookie-config>
+            <!-- A cookie with the HttpOnly attribute is inaccessible to the JavaScript Document.cookie API; it's only sent to the server. -->
             <http-only>true</http-only>
-            <secure>true</secure>
+            <!-- A cookie with the Secure attribute is only sent to the server with an encrypted request over the HTTPS protocol. 
+            It is strongly recommended to set this to true in a production environment; it may be set to false in an internal test environment
+            where HTTP is used instead of HTTPS. -->
+            <secure>${web.session.cookie.secure}</secure>
+            <!-- See more info on https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#restrict_access_to_cookies
+            and on on https://www.freecodecamp.org/news/session-hijacking-and-how-to-stop-it-711e3683d1ac/ -->
         </cookie-config>
     </session-config>
     <!--Configurations-->

--- a/sources/Re3gistry2Base/src/main/java/eu/europa/ec/re3gistry2/base/utility/Configuration.java
+++ b/sources/Re3gistry2Base/src/main/java/eu/europa/ec/re3gistry2/base/utility/Configuration.java
@@ -159,20 +159,28 @@ public class Configuration {
     }
 
     public void initServlet(HttpServletRequest request, HttpServletResponse response, boolean fromInstall, boolean fromInstalling) throws ServletException, IOException {
+        HttpSession session = request.getSession();
+        Configuration.getInstance().getLogger().debug("Start initialising servlet; session id: " + session.getId());
+        
         //Check installation
 
         // If the system is installed and the user tries to access the 
         // installation, he is redirected to the login
         if (checkInstallation() && (fromInstall || fromInstalling)) {
+          Configuration.getInstance().getLogger().trace("Redirecting to " + "." + WebConstants.PAGE_URINAME_LOGIN);
             response.sendRedirect("." + WebConstants.PAGE_URINAME_LOGIN);
         } else // If the system is installed and the user tries to access the 
         // installation, he is redirected to the login
         if (checkInstallRunning() && !fromInstalling) {
+          Configuration.getInstance().getLogger().trace("Redirecting to " + "." + WebConstants.PAGE_URINAME_INSTALL_RUNNING);
             response.sendRedirect("." + WebConstants.PAGE_URINAME_INSTALL_RUNNING);
         } else // If the system is not installed, and it is not installing every request is redirected to the installation
         if (!checkInstallation() && !checkInstallRunning() && !fromInstall) {
+          Configuration.getInstance().getLogger().trace("Redirecting to " + "." + WebConstants.PAGE_URINAME_INSTALL);
             response.sendRedirect("." + WebConstants.PAGE_URINAME_INSTALL);
         } else {
+            Configuration.getInstance().getLogger().trace("Proceed initialising servlet");
+          
             //Init properties
             Properties prop = Configuration.getInstance().getProperties();
             request.setAttribute(BaseConstants.KEY_REQUEST_PROPERTIES, prop);
@@ -182,30 +190,33 @@ public class Configuration {
             request.setAttribute(BaseConstants.KEY_REQUEST_LOCALIZATION, loc);
 
             //Init available languages in the session if needed
-            if (request.getSession().getAttribute(BaseConstants.KEY_SESSION_AVAILABLELANGUAGES) == null) {
+            if (session.getAttribute(BaseConstants.KEY_SESSION_AVAILABLELANGUAGES) == null) {
                 List<Localization> availableLanguages = new ArrayList<>(LocalizationMgr.getAvailableLanguages());
-                request.getSession().setAttribute(BaseConstants.KEY_SESSION_AVAILABLELANGUAGES, availableLanguages);
+                session.setAttribute(BaseConstants.KEY_SESSION_AVAILABLELANGUAGES, availableLanguages);
                 request.setAttribute(BaseConstants.KEY_SESSION_AVAILABLELANGUAGES, availableLanguages);
             } else {
-                request.setAttribute(BaseConstants.KEY_REQUEST_AVAILABLELANGUAGES, request.getSession().getAttribute(BaseConstants.KEY_SESSION_AVAILABLELANGUAGES));
+                request.setAttribute(BaseConstants.KEY_REQUEST_AVAILABLELANGUAGES, session.getAttribute(BaseConstants.KEY_SESSION_AVAILABLELANGUAGES));
             }
 
             String uri = request.getRequestURI();
             String currentPageName = "/" + uri.substring(uri.lastIndexOf("/") + 1);
             request.setAttribute(BaseConstants.KEY_REQUEST_CURRENTPAGENAME, currentPageName);
+            Configuration.getInstance().getLogger().debug("Current page name=" + currentPageName);
 
-            HttpSession session = request.getSession();
 
             //Getting logged user
             RegUser regUser = (RegUser) session.getAttribute(BaseConstants.KEY_SESSION_USER);
 
             if (regUser == null) {
+                Configuration.getInstance().getLogger().trace("No " + BaseConstants.KEY_SESSION_USER + " found in the session, checking user");
                 regUser = UserHelper.checkUser(request);
             }
 
             if (regUser != null) {
+                Configuration.getInstance().getLogger().trace("User found using checkUser: " + regUser);
                 request.setAttribute(BaseConstants.KEY_REQUEST_REGUSER, regUser);
             } else {
+                Configuration.getInstance().getLogger().trace("No user found");
                 if (!fromInstall && !fromInstalling) {
                     request.setAttribute(BaseConstants.KEY_REQUEST_REGUSER, null);
                     //request.getRequestDispatcher("/jsp/login.jsp").forward(request, response);

